.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: Tag on Github and Release to RubyGems
+on: [workflow_dispatch]
+
+jobs:
+  release:
+    name: Release to RubyGems
+    runs-on: ubuntu-latest
+    # Ideally, we would require successful tests before
+    # allowing to publish. Because the tests need to be
+    # in a separate workflow file, we can't reference it
+    # here.
+    # needs: test
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+      - name: Tag on Github
+        run: |
+          # Todo: maybe there's a better way to get the version
```

Comment:
If you have some advice here, let me know

Reply:
I can only think of an indirect way that's higher level:

```ruby
puts Gem::Specification::load("keycloak_oauth.gemspec").version
```

Reply:
I'm now using this, thanks